src/rivescript.coffee

```diff
@@ -755,4 +763,19 @@ class RiveScript
   reply: (user, msg, scope) ->
     return @brain.reply(user, msg, scope)
 
+  ##
+  # Promise replyAsync (string username, string message [[, scope], callback])
+  #
+  # Asyncronous version of reply. Use replyAsync if at least one of the subroutines
+  # used with <call> tag returns a promise
```

Comment:
Add documentation for the `callback` argument. It looks from the source that it should be a function that accepts `(error, reply)` as parameters?

Also provide examples of both use cases, like

```coffeescript
# Example: using promises
#
# ```javascript
# rs.replyAsync(user, message, this).then(function(error, reply) {
#     console.log("Bot>", reply);
# });
# ```
#
# Example: using the callback
#
# ```javascript
# rs.replyAsync(username, msg, this, function(error, reply) {
#     console.log("Bot>", reply);
# });
# ```
```

(the documentation uses GitHub flavored markdown)